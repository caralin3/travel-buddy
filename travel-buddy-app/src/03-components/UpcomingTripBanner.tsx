import moment from 'moment';
import React from 'react';
import { Col, Row } from 'reactstrap';
import { Banner } from '../01-atoms';
import { Countdown } from '../02-molecules';

export interface UpcomingTripBannerProps {
  date: string;
  title: string;
}

export const UpcomingTripBanner: React.FC<UpcomingTripBannerProps> = ({ date, title }) => (
  <Banner>
    <Row>
      <Col>
        <p className="text-uppercase">
          <strong>Upcoming Trip</strong>
        </p>
      </Col>
    </Row>
    <Row>
      <Countdown date={date} />
    </Row>
    <Row>
      <Col>
        <h2>{title}</h2>
        <p className="m-0">
          <strong>{moment(date).format('MMMM DD, YYYY')}</strong>
        </p>
      </Col>
    </Row>
  </Banner>
);
