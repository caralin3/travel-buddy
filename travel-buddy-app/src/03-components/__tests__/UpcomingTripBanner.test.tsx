import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UpcomingTripBanner } from '../UpcomingTripBanner';

describe('UpcomingTripBanner', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UpcomingTripBanner title="Trip" date="2121-05-12" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
