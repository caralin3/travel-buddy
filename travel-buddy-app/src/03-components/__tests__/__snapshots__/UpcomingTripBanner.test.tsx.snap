// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingTripBanner should render correctly 1`] = `
<Banner>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <p
        className="text-uppercase"
      >
        <strong>
          Upcoming Trip
        </strong>
      </p>
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Countdown
      date="2121-05-12"
    />
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <h2>
        Trip
      </h2>
      <p
        className="m-0"
      >
        <strong>
          May 12, 2121
        </strong>
      </p>
    </Col>
  </Row>
</Banner>
`;
